src/aerialmap_display.cpp

```diff
@@ -533,39 +561,32 @@ void AerialMapDisplay::transformAerialMap()
   auto const originOffsetX = center.x - std::floor(center.x);
   auto const originOffsetY = 1 - center.y + std::floor(center.y);  // y coord is flipped
   double const tile_w_h_m = getTileWH();
-  position.x -= originOffsetX * tile_w_h_m;
-  position.y -= originOffsetY * tile_w_h_m;
-  scene_node_->setPosition(position);
 
-  int const convention = frame_convention_property_->getOptionInt();
-  if (convention == FRAME_CONVENTION_XYZ_ENU)
-  {
-    // ENU corresponds to our default drawing method
-    scene_node_->setOrientation(Ogre::Quaternion::IDENTITY);
-  }
-  else if (convention == FRAME_CONVENTION_XYZ_NED)
-  {
-    // XYZ->NED will cause the map to appear reversed when viewed from above (from +z).
-    // clang-format off
-		Ogre::Matrix3 const xyz_R_ned(0, 1, 0,
-		                              1, 0, 0,
-		                              0, 0, -1);
-    // clang-format on
-    scene_node_->setOrientation(xyz_R_ned.Transpose());
-  }
-  else if (convention == FRAME_CONVENTION_XYZ_NWU)
-  {
-    // clang-format off
-		Ogre::Matrix3 const xyz_R_nwu(0, -1, 0,
-		                              1, 0, 0,
-		                              0, 0, 1);
-    // clang-format on
-    scene_node_->setOrientation(xyz_R_nwu.Transpose());
-  }
-  else
+  if (!context_->getFrameManager()->getTransform(frameMap, ref_fix_.header.stamp, positionUnused, orientationOffset))
   {
-    ROS_ERROR_STREAM("Invalid convention code: " << convention);
+    // display error
+    std::string error;
+    if (context_->getFrameManager()->transformHasProblems(frameMap, ref_fix_.header.stamp, error))
+    {
+      setStatus(StatusProperty::Error, "Transform", QString::fromStdString(error));
+    }
+    else
+    {
+      setStatus(StatusProperty::Error, "Transform",
+                "Could not transform from [" + QString::fromStdString(frameMap) + "] to Fixed Frame [" + fixed_frame_ +
+                    "] for an unknown reason");
+    }
   }
+
+  Ogre::Vector3 offsetMap =
```

Comment:
Since all the code is not self-explaining, I'm missing extensive inline-comments on those transform:

```
# lookup rotation from A to B
# rotate foo by B in order to blah
# now translate by baz such that the map is there w.r.t. frame A
...
``` 
Despite that, variable names could be more clear (like `orientation_targetFrame_sourceFrame`)
From reading the code, its rather unclear what the problem is, and how the code solves it.

Problem should be something like:
- gps is reported w.r.t. gps sensor frame
- the tiles retrieved have a fixed center position
- the reported gps position is somewhere within the retrieved tile coordinate system
- hence we need to transform the scene-node, such that .. 
- ...